java/java.source.base/src/org/netbeans/modules/java/source/save/CasualDiff.java

```diff
@@ -1966,8 +1967,23 @@ protected int diffBindingPattern(JCBindingPattern oldT, JCBindingPattern newT, i
 
     protected int diffCase(JCCase oldT, JCCase newT, int[] bounds) {
         int localPointer = bounds[0];
-        List<JCExpression> oldPatterns = getCasePatterns(oldT);
-        List<JCExpression> newPatterns = getCasePatterns(newT);
+        List<? extends JCTree> oldPatterns;
+        List<? extends JCTree> newPatterns;
+        
+        if(TreeShims.isJDKVersionRelease17_Or_Above() && 
```

Comment:
This should always be true currently, right? I mean, NetBeans no longer supports javac API older that 17 (in NetBeans 13) or 18 (in the mainline).

Reply:
Thanks for the input. have adressed the comments and removed reflection access wherever possible.